wgsl/index.bs

```diff
@@ -9805,13 +9808,22 @@ The rules for analyzing expressions take as argument both the expression itself
       <td class="nowrap">*CF*, *CF*
       <td>
    <tr><td>identifier [=resolves|resolving=] to function-scope variable "x"
+      where the [=load rule=] is invoked during [=type checking=] with "x" as
+      [=root identifier=] of the [=memory view=]
       <td>*Result*
       <td class>*X* is the node corresponding to the value of "x" at the input to the statement containing this expression
       <td class="nowrap">*CF*, *Result*
       <td class>*Result* -> {*CF*, *X*}
 
       Note: *X* is equivalent to *Vout*(*prev*) for "x"<br>
       (see [[#func-var-value-analysis]])
+   <tr><td>identifier [=resolves|resolving=] to function-scope variable "x"
```

Comment:
This is a bit wobbly for me.  
The row is matched when we're looking at an identifier.
But the thing that  the load rule is acting on is at  a much higher level.

Maybe wordsmithing it helps:

> identifier [=resolves|resolving=] to function-scope variable "x", where the identifier appears as the root identifier of a memory view expression MVE, and the load rule is not invoked on MVE during type checking.

Even this is slightly weird because memory view expressions nest. But in the end it doesn't matter because the load rule only occurs at the outer most level. (heh, because we can't load or store pointers yet...)

So if you pick wrong and look at it too low-level, the CF,CF is harmless, just passing through the control flow uniformity properties onto the value. Which still fits.

Reply:
I suppose another problem is that there is an implied order of events here. When is type checking performed compared to the uniformity analysis? The type checking only works for this if desugaring occurs first, but I wouldn't expect a tool to output the desugared version of the module.

Reply:
I've adopted that language and added a note in desugaring about type checking occurring multiple times.